File: app/src/main/java/com/k4dima/chat/core/repository/LemonadeRepository.java
package com.k4dima.chat.core.repository;

import android.support.annotation.NonNull;

import com.k4dima.chat.core.model.Message;
import com.k4dima.chat.core.model.ResponseMessages;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;

public interface LemonadeRepository {
    @NonNull
    Observable<ResponseMessages> message(Message message);

    Single<List<Message>> saved();
}

File: app/src/main/java/com/k4dima/chat/features/messages/repository/LemonadeRepository.kt
package com.k4dima.chat.features.messages.repository

import com.k4dima.chat.core.model.Message
import com.k4dima.chat.core.model.ResponseMessages

import io.reactivex.Single

interface LemonadeRepository {
    fun message(message: Message): Single<ResponseMessages>

    fun saved(): Single<MutableList<Message>>
}